modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/ObjectFieldFormBase.tsx

```diff
@@ -303,7 +305,14 @@ export default function ObjectFieldFormBase({
 				: false;
 		}
 
-		return disabled || values.localized || values.state;
+		return !!(
+			(objectDefinition?.status?.label === 'approved' &&
+				!objectFieldRequired &&
+				values.id) ||
```

Comment:
Hey @igor-franca, Hey Igor, as we discussed, you can extract this logic to a specific `if` above and use the `isApproved` and `editObjectField` props instead of the `objectDefinition?.status?.label === 'approved'` and `values.id` respectively